code/nebula/neblightning.cpp

```diff
@@ -152,122 +152,245 @@ DCF(lightning_intensity, "Sets lightning intensity between 0.0 and 1.0 (Default
+				// Don't create bolt if it has +nocreate and is in a modular table.
+				if (!create_if_not_found && Parsing_modular_table) {
+					if (!skip_to_start_of_string_either("$Bolt:", "#Bolts end")) {
+						error_display(1, "Missing [#Bolts end] or [$Bolt] after lightning bolt %s", bolt_t.name);
+					}
+					continue;
+				}
+
+				Bolt_types.push_back(bolt_t);
+				bolt_p = &Bolt_types[Bolt_types.size() - 1];
+			}
 
 			// b_scale
-			required_string("+b_scale:");
-			stuff_float(&new_bolt_type.b_scale);
+			if (optional_string("+b_scale:")) {
+				stuff_float(&bolt_p->b_scale);
+			}
 
 			// b_shrink
-			required_string("+b_shrink:");
-			stuff_float(&new_bolt_type.b_shrink);
+			if (optional_string("+b_shrink:")) {
+				stuff_float(&bolt_p->b_shrink);
+			}
 
 			// b_poly_pct
-			required_string("+b_poly_pct:");
-			stuff_float(&new_bolt_type.b_poly_pct);
+			if (optional_string("+b_poly_pct:")) {
+				stuff_float(&bolt_p->b_poly_pct);
+			}
 
-			// child rand
-			required_string("+b_rand:");
-			stuff_float(&new_bolt_type.b_rand);
+			// child rand, omg not the chilren!
+			if (optional_string("+b_rand:")) {
+				stuff_float(&bolt_p->b_rand);
+			}
 
 			// z add
-			required_string("+b_add:");
-			stuff_float(&new_bolt_type.b_add);
+			if (optional_string("+b_add:")) {
+				stuff_float(&bolt_p->b_add);
+			}
 
 			// # strikes
-			required_string("+b_strikes:");
-			stuff_int(&new_bolt_type.num_strikes);
+			if (optional_string("+b_strikes:")) {
+				stuff_int(&bolt_p->num_strikes);
+			}
 
 			// lifetime
-			required_string("+b_lifetime:");
-			stuff_int(&new_bolt_type.lifetime);
+			if (optional_string("+b_lifetime:")) {
+				stuff_int(&bolt_p->lifetime);
+			}
 
 			// noise
-			required_string("+b_noise:");
-			stuff_float(&new_bolt_type.noise);
+			if (optional_string("+b_noise:")) {
+				stuff_float(&bolt_p->noise);
+			}
 
 			// emp effect
-			required_string("+b_emp:");
-			stuff_float(&new_bolt_type.emp_intensity);
-			stuff_float(&new_bolt_type.emp_time);
+			if (optional_string("+b_emp:")) {
+				stuff_float(&bolt_p->emp_intensity);
+				stuff_float(&bolt_p->emp_time);
+			}
 
 			// texture
-			required_string("+b_texture:");
-			stuff_string(name, F_NAME, sizeof(name));
-			if (!Fred_running){
-				new_bolt_type.texture = bm_load(name);
+			if (optional_string("+b_texture:")) {
+				stuff_string(name, F_NAME, sizeof(name));
+				if (!Fred_running) {
+					bolt_p->texture = bm_load(name);
+				}
 			}
 
+			if (bolt_p->texture < 0)
+				error_display(1, "Bolt %s has no texture defined.", bolt_p->name);
+
 			// glow
-			required_string("+b_glow:");
-			stuff_string(name, F_NAME, sizeof(name));
-			if (!Fred_running){
-				new_bolt_type.glow = bm_load(name);
+			if (optional_string("+b_glow:")) {
+				stuff_string(name, F_NAME, sizeof(name));
+				if (!Fred_running) {
+					bolt_p->glow = bm_load(name);
+				}
 			}
 
+			if (bolt_p->glow < 0)
+				error_display(1, "Bolt %s has no glow texture defined.", bolt_p->name);
```

Comment:
No textures makes sense, but is no glow necessarily a problem?

Reply:
You know.. now that you ask, I'm not sure the glow texture is even used anywhere. At least I can't seem to find it. I added this check here because it was required by the original parsing code.

Reply:
OK, I did a test with a build that completely removed `int glow` from the bolt struct and threw away the data if parsed. Things seemed to work without issue or error in the several retail nebula missions I tested. Reading the code it looks like the glow is actually determined by the `+b_add` value. So it would probably make sense to deprecate the glow texture completely, but I'd feel more comfortable doing that if someone could double check my findings.

Reply:
It's fine, no need to go far (at least on this PR), I'd just remove the warning for no glow.

Reply:
Fair enough. Done.